lms/views/decorators/legacy_h_api.py

```diff
@@ -62,16 +62,20 @@ def wrapper(request, jwt, context=None):
             ],
         }
 
-        # Send a PATCH user request to update user data to keep it in sync
-        # with the data sent over LTI. This will succeed if the user already
-        # exists.
         try:
-            hapi_svc.patch(
-                f"users/{context.h_username}", {"display_name": context.h_display_name}
-            )
-        except HAPINotFoundError:
-            # Call the h API to create the user in h if it doesn't exist already.
-            hapi_svc.post("users", user_data)
+            # Send a PATCH user request to update user data to keep it in sync
+            # with the data sent over LTI. This will succeed if the user already
+            # exists.
+            try:
+                hapi_svc.patch(
+                    f"users/{context.h_username}",
+                    {"display_name": context.h_display_name},
+                )
+            except HAPINotFoundError:
+                # Call the h API to create the user in h if it doesn't exist already.
+                hapi_svc.post("users", user_data)
+        except HAPIError as err:
+            raise HTTPInternalServerError(explanation=err.explanation) from err
```

Comment:
I'm having to make the same changes twice because we have `h_api` view decorators and `legacy_h_api` ones. The legacy ones will be deleted soon